### PR TITLE
fix conflict multiple update

### PR DESCRIPTION
**Pull request by ckr123**

### CODE REVIEW

File: domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs [hunk 1]
```diff
@@ -60,8 +61,11 @@ internal class ContractServiceImpl : IContractService
                 return new GsrnNotFound();
             }
 
-            var overlappingContract = contractsByGsrn.FirstOrDefault(c =>
+            var contractsGsrn = contractsByAllGsrn.Where(c => c.GSRN == contract.GSRN).ToList();
```
Comment (martinhenningjensen) [inline domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs:64]: What happens if the request contains multiple contracts on the same GSRN number? 

It looks like contractsByAllGsrn variable is not updated with contracts created as part of the request handling?

File: domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs [hunk 2]
```diff
@@ -161,7 +164,6 @@ private static MeteringPointType Map(MeterType type)
         var contractsByGsrn =
             await unitOfWork.CertificateIssuingContractRepo.GetByGsrn(issuingContracts.Select(c => c.GSRN).ToList(),
                 cancellationToken);
-
         foreach (var updatedContract in contracts.Contracts)
```
Comment (martinhenningjensen) [inline domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs:167]: Does this have the same issue as mentioned previously?

**Review by martinhenningjensen — CHANGES_REQUESTED** [review]

File: domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs (hunk 2 above)
Comment (ckr123) [reply domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs:177]: fixed in 1eb3207b

File: domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs (hunk 1 above)
Comment (ckr123) [reply domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs]: fixed in 1eb3207b

Comment (martinhenningjensen) [reply domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs]: I don't see a test where multiple contracts with same GSRN are bulk-created :)

**Review by martinhenningjensen — APPROVED** [review]

**Pull request merged**